runtime/vm/resolvesupport.cpp

```diff
@@ -1839,6 +1839,101 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = ramCPEntry->value;
+	J9JavaVM *vm = vmThread->javaVM;
+	BOOLEAN resolved = FALSE;
+
+retry:
+	/* Check if already resolved */
+	if (NULL != value) {
+		resolved = TRUE;
+	} else if (NULL != ramCPEntry->exception) {
+		J9Class *throwable = J9VMJAVALANGTHROWABLE_OR_NULL(vm);
+		if (isSameOrSuperClassOf(throwable, J9VM_J9CLASS_FROM_HEAPCLASS(vmThread, ramCPEntry->exception))) {
+			/* Resolved with exception */
+			VM_VMHelpers::setExceptionPending(vmThread, ramCPEntry->exception);
+			resolved = TRUE;
+		} else if (ramCPEntry->exception == vm->voidReflectClass->classObject) {
+			/* Resolved with null reference */
+			resolved = TRUE;
+		} else if (ramCPEntry->exception != vmThread->threadObject) {
+			/* Another thread is currently trying to resolve the entry */
+			VM_AtomicSupport::yieldCPU();
```

Comment:
@DanHeidinga @gacholio In the case that another thread is currently resolving the entry, would thread yield be sufficient to wait until the resolving thread finish?

Reply:
No, yield is essentially a no-op - it guarantees nothing.

Reply:
sry, I meant if yield is used here, would it be sufficient (as in suspending the thread for short period) so it will not be consistently spinning in the `goto retry;` loop as other thread is resolving the condy entry. (so we don't have this thread wasting resources while not doing any actual work, a mutex would be ideal but too expensive?)